### PR TITLE
Run Spacefinder on deadblogs

### PR DESCRIPTION
**Pull request by chrislomaxjones**
## What does this change?

Add support for running Spacefinder on DCR deadblogs. The code in the commercial bundle already runs on these pages, this PR adds the hints necessary to insert inline ad slots between liveblocks.

These hints take the form of two added classnames:
- `js-liveblog-body` applied to `ArticleBody` when rendering a liveblog/deadblog. This represents the container in which Spacefinder will attempt to insert inline ads.
- `block` applied to each of the `Liveblock` components. Spacefinder will select for these as the candidates before which inline ads can be inserted. Candidates are the direct descendants of the container above.

Additionally, the adverts that are inserted into liveblogs by Spacefinder require some additional styling. These styles are added to the `ArticleContainer` along with other dynamic ad styles, and scoped to target the classes of slots added by Spacefinder.

### Before

Screenshot 2022-01-07 at 09 26 38

### After

Screenshot 2022-01-07 at 09 26 11

## Why?

As migration to DCR continues it's necessary to also migrate the ability to include inline ads on these new pages.

### CODE REVIEW

File: common-rendering/src/components/liveBlockContainer.tsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { css } from "@emotion/react";
-import { neutral, from, space, headline } from '@guardian/source-foundations';
-import { FirstPublished } from './FirstPublished';
+import { neutral, from, space, headline } from "@guardian/source-foundations";
+import { FirstPublished } from "./FirstPublished";
```
Comment (OllysCoding) [inline common-rendering/src/components/liveBlockContainer.tsx:3]: I think we lack any code-style checks in common-rendering which is why we're seeing changes like this (which isn't ideal) - perhaps something we need to look at outside this team

**Review by oliverlloyd — COMMENTED** [review]
This change looks good in general. Is there any help you need progressing this to Ready for Review?

The only risk I can think of here is that the scope of the css might bleed but it looks to me like these changes are limited to specific classnames

File: common-rendering/src/components/liveBlockContainer.tsx (hunk 1 above)
Comment (chrislomaxjones) [reply common-rendering/src/components/liveBlockContainer.tsx:3]: I see, I think this is auto-formatting that I hadn't noticed then. If it's simpler I can remove the formatting changes from this PR and possibly address them separately (run prettier in common-rendering in a separate PR maybe?)

**Comment by chrislomaxjones** [comment]
> This change looks good in general. Is there any help you need progressing this to Ready for Review?

I'm currently still nailing down the styles for outstream ads, hopefully that won't be too much of a problem though!

> The only risk I can think of here is that the scope of the css might bleed but it looks to me like these changes are limited to specific classnames

Once I've got the styling figured out I think the selectors should all just be classnames, with the elements matching those classnames being inserted by Spacefinder.

File: common-rendering/src/components/liveBlockContainer.tsx (hunk 1 above)
Comment (oliverlloyd) [reply common-rendering/src/components/liveBlockContainer.tsx:3]: I think it's fine to keep them here. They're small and self contained. We're working on addressing the linting issues for common in a separate thread

Comment (chrislomaxjones) [reply common-rendering/src/components/liveBlockContainer.tsx:3]: Alright, sounds good!

File: dotcom-rendering/docs/contracts/001-commercial-selectors.md [hunk 2]
```diff
@@ -2,7 +2,10 @@
 
 ## What is the contract?
 
-We place the `article-body-commercial-selector` class on the container element of the article body. (TODO: are there others?)
+We place the following classes on the container element of the article body:
+
+-   `article-body-commercial-selector` on the ArticleRenderer
+-   `js-liveblog-body` on the ArticleBody when rendering a Liveblog/Deadblog
```
Comment (oliverlloyd) [inline dotcom-rendering/docs/contracts/001-commercial-selectors.md:8]: Thank you for including this change! 🙏

File: dotcom-rendering/src/web/components/ArticleContainer.tsx [hunk 3]
```diff
@@ -109,6 +109,42 @@ const articleAdStyles = css`
...
+		margin: 0 auto ${space[3]}px;
+
+		.ad-slot__label {
+			color: ${neutral[46]};
+			border-top-color: ${neutral[86]};
+		}
+
+		&.ad-slot--outstream {
+			${from.tablet} {
+				width: 620px;
```
Comment (oliverlloyd) [inline dotcom-rendering/src/web/components/ArticleContainer.tsx:122]: I'm not certain what this width is doing but liveblogs have a width of `700px`, not `620px` - does this need to be accounted for here as well?

See: https://github.com/guardian/dotcom-rendering/blob/oliver/instagram-island/dotcom-rendering/src/web/layouts/LiveLayout.tsx#L175

File: common-rendering/src/components/liveBlockContainer.tsx [hunk 4]
```diff
@@ -52,6 +52,7 @@ const LiveBlockContainer = ({
 		<article
 			id={`block-${id}`}
 			key={id}
+			className="block"
```
Comment (oliverlloyd) [inline common-rendering/src/components/liveBlockContainer.tsx:55]: The comment included in `ArticleBody` was great, can we do something similar here as well?

**Review by oliverlloyd — APPROVED** [review]
This looks really great.

I've made a comment about a reference to 620 pixels I saw in the code but liveblogs are 700 pixels wide which might need checking and a teeny ask for a comment but I'm happy to approve this work and leave it up to you to work out the best approach for these comments

File: dotcom-rendering/src/web/components/ArticleContainer.tsx (hunk 3 above)
Comment (chrislomaxjones) [reply dotcom-rendering/src/web/components/ArticleContainer.tsx:122]: This width is to account for [outstream ads](https://github.com/guardian/commercial-core/blob/main/src/ad-sizes.ts#L70) which have a width of `620px`.

Comment (oliverlloyd) [reply dotcom-rendering/src/web/components/ArticleContainer.tsx:122]: > This width is to account for [outstream ads](https://github.com/guardian/commercial-core/blob/main/src/ad-sizes.ts#L70) which have a width of `620px`.

Ah, I see. Thanks for checking!

File: common-rendering/src/components/liveBlockContainer.tsx (hunk 4 above)
Comment (chrislomaxjones) [reply common-rendering/src/components/liveBlockContainer.tsx:56]: Sure, I've added a comment!

**Pull request merged**